src/editor/ui_menus/main_menu_map_options.cc

```diff
@@ -36,6 +36,146 @@ inline EditorInteractive& MainMenuMapOptions::eia() {
 	return dynamic_cast<EditorInteractive&>(*get_parent());
 }
 
+constexpr unsigned kSuggestedTeamsUnitSize = 24;
+
+SuggestedTeamsEntry::SuggestedTeamsEntry(MainMenuMapOptions* mmmo,
+                                         UI::Panel* parent,
+                                         const Widelands::Map& map,
+                                         unsigned w,
+                                         Widelands::SuggestedTeamLineup t)
+   : UI::Panel(parent, 0, 0, w, kSuggestedTeamsUnitSize, _("Click player to remove")),
+     map_(map),
+     team_(t),
+     delete_(this,
+             "delete",
+             0,
+             0,
+             kSuggestedTeamsUnitSize,
+             kSuggestedTeamsUnitSize,
+             UI::ButtonStyle::kWuiSecondary,
+             _("Delete"),
+             _("Delete this suggested team lineup")) {
+	for (size_t index = 0; index < team_.size(); ++index) {
+		dropdowns_.push_back(create_dropdown(index));
+	}
+
+	delete_.sigclicked.connect([this, mmmo]() { mmmo->delete_suggested_team(this); });
+
+	update();
+}
+
+UI::Dropdown<Widelands::PlayerNumber>* SuggestedTeamsEntry::create_dropdown(size_t index) {
+	UI::Dropdown<Widelands::PlayerNumber>* dd = new UI::Dropdown<Widelands::PlayerNumber>(
+	   this, std::to_string(index), 0, index * kSuggestedTeamsUnitSize, kSuggestedTeamsUnitSize, 8,
+	   0, _("+"), UI::DropdownType::kPictorialMenu, UI::PanelStyle::kWui,
+	   UI::ButtonStyle::kWuiSecondary);
+	for (size_t i = 0; i < map_.get_nrplayers(); ++i) {
+		dd->add(map_.get_scenario_player_name(i + 1), i,
+		        playercolor_image(i, "images/players/player_position_menu.png"));
+	}
+	dd->set_tooltip(_("Add a player to this team"));
+	dd->selected.connect([this, dd]() {
+		const Widelands::PlayerNumber player = dd->get_selected();
+		// add this player to this team and remove him from all other teams in this lineup
+		for (Widelands::SuggestedTeam& t : team_) {
+			for (auto it = t.begin(); it != t.end(); ++it) {
+				if (*it == player) {
+					t.erase(it);
+					break;
+				}
+			}
+		}
+		// determine our index (it may have changed since the creation!)
+		int dd_index = -1;
+		for (size_t i = 0; i < dropdowns_.size(); ++i) {
+			if (dropdowns_[i] == dd) {
+				dd_index = i;
+				break;
+			}
+		}
+		assert(dd_index >= 0);
```

Comment:
Why not simply initialize it with 0? The asserts only catch illegal state in debug builds and we can end up with weird behaviour in release builds.

If you want add some checks, you could add a boolean flag and simply check it.

Reply:
ok